### PR TITLE
fix(components): [select] support custom height (#10840)

### LINKED ISSUE #10840: [Component] [select] el-select的disabled属性切换时，如果自定义了line-height，里面input框的高度计算有问题
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.25`
- Browser / OS: `edge 104.0.1293.70`
- Build Tool: `Vite`

### Related Component

- `el-select`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJycpXG5jb25zdCBzZWxlY3REaXNhYmxlZCA9IHJlZihmYWxzZSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxkaXYgY2xhc3M9XCJjb250YWluZXJcIj5cbiAgICDlvIDlkK8v5YWz6Zetc2VsZWN0XG4gICAgPGVsLXN3aXRjaCB2LW1vZGVsPVwic2VsZWN0RGlzYWJsZWRcIiAvPlxuICAgIDxlbC1zZWxlY3Qgdi1tb2RlbD1cIm1zZ1wiIDpkaXNhYmxlZD1cInNlbGVjdERpc2FibGVkXCI+XG4gICAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiMVwiIGxhYmVsPVwiMVwiPjwvZWwtb3B0aW9uPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjJcIj48L2VsLW9wdGlvbj5cbiAgICA8L2VsLXNlbGVjdD5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c3R5bGU+XG4gIC5jb250YWluZXIgLmVsLWlucHV0IHtcbiAgICAtLWVsLWlucHV0LWhlaWdodDogNDJweDtcbiAgfVxuPC9zdHlsZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

1.将el-input的css变量--el-input-height置为42px;
2.改变el-select的disabled属性
3.会发现input的高度被强制设为了30px，但是我line-height设置为42px，所以应该计算为40px
image
yWDevc9gQF
image

### What is Expected?

input高度计算为40px，或者不设置height属性

### What is actually happening?

input高度计算为30px，导致变矮

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by MrWeilian**
Fix: [[Component] [select] el-select的disabled属性切换时，如果自定义了line-height，里面input框的高度计算有问题](https://github.com/element-plus/element-plus/issues/10840)
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by MrWeilian** [comment]
[Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cxID0gcmVmKCcnKVxuY29uc3QgbXNnMiA9IHJlZignJylcbmNvbnN0IG1zZzMgPSByZWYoJycpXG5jb25zdCBtc2c0ID0gcmVmKCcnKVxuY29uc3Qgc2VsZWN0RGlzYWJsZWQgPSByZWYoZmFsc2UpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuXHQ8cD7lvIDlkK8v5YWz6Zetc2VsZWN077yaPGVsLXN3aXRjaCB2LW1vZGVsPVwic2VsZWN0RGlzYWJsZWRcIiAvPjwvcD5cbiAgPGRpdiBjbGFzcz1cImNvbnRhaW5lclwiPlxuICAgIDxoMj7nlKjmiLfoh6rlrprkuYnpq5jluqbvvIzku6XnlKjmiLflrprkuYnkuLrlh4Y8L2gyPlxuICAgIDxlbC1zZWxlY3Qgdi1tb2RlbD1cIm1zZzFcIiA6ZGlzYWJsZWQ9XCJzZWxlY3REaXNhYmxlZFwiPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjFcIiBsYWJlbD1cIjExMTExMTFcIj48L2VsLW9wdGlvbj5cbiAgICAgIDxlbC1vcHRpb24gdmFsdWU9XCIyXCIgbGFiZWw9XCIyMjIyMjIyXCI+PC9lbC1vcHRpb24+XG4gICAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiM1wiIGxhYmVsPVwiMzMzMzMzM1wiPjwvZWwtb3B0aW9uPlxuICAgIDwvZWwtc2VsZWN0PlxuICAgIDxlbC1zZWxlY3Qgdi1tb2RlbD1cIm1zZzJcIiA6ZGlzYWJsZWQ9XCJzZWxlY3REaXNhYmxlZFwiIG11bHRpcGxlIGNsZWFyYWJsZT5cbiAgICAgIDxlbC1vcHRpb24gdmFsdWU9XCIxXCIgbGFiZWw9XCIxMTExMTExXCI+PC9lbC1vcHRpb24+XG4gICAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiMlwiIGxhYmVsPVwiMjIyMjIyMlwiPjwvZWwtb3B0aW9uPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjNcIiBsYWJlbD1cIjMzMzMzMzNcIj48L2VsLW9wdGlvbj5cbiAgICA8L2VsLXNlbGVjdD5cbiAgICA8ZWwtc2VsZWN0IHYtbW9kZWw9XCJtc2cyXCIgOmRpc2FibGVkPVwic2VsZWN0RGlzYWJsZWRcIiBtdWx0aXBsZSBjbGVhcmFibGUgc2l6ZT1cInNtYWxsXCI+XG4gICAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiMVwiIGxhYmVsPVwiMTExMTExMVwiPjwvZWwtb3B0aW9uPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjIyMjIyMjJcIj48L2VsLW9wdGlvbj5cbiAgICAgIDxlbC1vcHRpb24gdmFsdWU9XCIzXCIgbGFiZWw9XCIzMzMzMzMzXCI+PC9lbC1vcHRpb24+XG4gICAgPC9lbC1zZWxlY3Q+XG4gICAgPGVsLXNlbGVjdCB2LW1vZGVsPVwibXNnMlwiIDpkaXNhYmxlZD1cInNlbGVjdERpc2FibGVkXCIgbXVsdGlwbGUgY2xlYXJhYmxlIHNpemU9XCJsYXJnZVwiPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjFcIiBsYWJlbD1cIjExMTExMTFcIj48L2VsLW9wdGlvbj5cbiAgICAgIDxlbC1vcHRpb24gdmFsdWU9XCIyXCIgbGFiZWw9XCIyMjIyMjIyXCI+PC9lbC1vcHRpb24+XG4gICAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiM1wiIGxhYmVsPVwiMzMzMzMzM1wiPjwvZWwtb3B0aW9uPlxuICAgIDwvZWwtc2VsZWN0PlxuICAgIDxlbC1zZWxlY3Qgdi1tb2RlbD1cIm1zZzJcIiA6ZGlzYWJsZWQ9XCJzZWxlY3REaXNhYmxlZFwiIG11bHRpcGxlIGNsZWFyYWJsZSBzaXplPVwiZGVmYXVsdFwiPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjFcIiBsYWJlbD1cIjExMTExMTFcIj48L2VsLW9wdGlvbj5cbiAgICAgIDxlbC1vcHRpb24gdmFsdWU9XCIyXCIgbGFiZWw9XCIyMjIyMjIyXCI+PC9lbC1vcHRpb24+XG4gICAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiM1wiIGxhYmVsPVwiMzMzMzMzM1wiPjwvZWwtb3B0aW9uPlxuICAgIDwvZWwtc2VsZWN0PlxuICA8L2Rpdj5cbiAgPGgyPuaXoOeUqOaIt+WumuS5iemrmOW6pu+8jOS7pXNpemXjgIHpu5jorqRzaXpl5Li65YeG44CC5aaC77yac2l6ZT1zbWFsbDwvaDI+XG4gIDxlbC1zZWxlY3Qgdi1tb2RlbD1cIm1zZzNcIiA6ZGlzYWJsZWQ9XCJzZWxlY3REaXNhYmxlZFwiIHNpemU9XCJzbWFsbFwiPlxuICAgIDxlbC1vcHRpb24gdmFsdWU9XCIxXCIgbGFiZWw9XCIxMTExMTExXCI+PC9lbC1vcHRpb24+XG4gICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjIyMjIyMjJcIj48L2VsLW9wdGlvbj5cbiAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiM1wiIGxhYmVsPVwiMzMzMzMzM1wiPjwvZWwtb3B0aW9uPlxuICA8L2VsLXNlbGVjdD5cbiAgPGVsLXNlbGVjdCB2LW1vZGVsPVwibXNnNFwiIDpkaXNhYmxlZD1cInNlbGVjdERpc2FibGVkXCIgc2l6ZT1cInNtYWxsXCIgbXVsdGlwbGUgY2xlYXJhYmxlPlxuICAgIDxlbC1vcHRpb24gdmFsdWU9XCIxXCIgbGFiZWw9XCIxMTExMTExXCI+PC9lbC1vcHRpb24+XG4gICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjIyMjIyMjJcIj48L2VsLW9wdGlvbj5cbiAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiM1wiIGxhYmVsPVwiMzMzMzMzM1wiPjwvZWwtb3B0aW9uPlxuICA8L2VsLXNlbGVjdD5cbiAgPGgyPuaXoOeUqOaIt+WumuS5iemrmOW6pu+8jOS7pXNpemXjgIHpu5jorqRzaXpl5Li65YeG44CC5aaC77ya5LiN6K6+572uc2l6ZTwvaDI+XG4gIDxlbC1zZWxlY3Qgdi1tb2RlbD1cIm1zZzNcIiA6ZGlzYWJsZWQ9XCJzZWxlY3REaXNhYmxlZFwiPlxuICAgIDxlbC1vcHRpb24gdmFsdWU9XCIxXCIgbGFiZWw9XCIxMTExMTExXCI+PC9lbC1vcHRpb24+XG4gICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjIyMjIyMjJcIj48L2VsLW9wdGlvbj5cbiAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiM1wiIGxhYmVsPVwiMzMzMzMzM1wiPjwvZWwtb3B0aW9uPlxuICA8L2VsLXNlbGVjdD5cbiAgPGVsLXNlbGVjdCB2LW1vZGVsPVwibXNnNFwiIDpkaXNhYmxlZD1cInNlbGVjdERpc2FibGVkXCIgbXVsdGlwbGUgY2xlYXJhYmxlPlxuICAgIDxlbC1vcHRpb24gdmFsdWU9XCIxXCIgbGFiZWw9XCIxMTExMTExXCI+PC9lbC1vcHRpb24+XG4gICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjIyMjIyMjJcIj48L2VsLW9wdGlvbj5cbiAgICA8ZWwtb3B0aW9uIHZhbHVlPVwiM1wiIGxhYmVsPVwiMzMzMzMzM1wiPjwvZWwtb3B0aW9uPlxuICA8L2VsLXNlbGVjdD5cblxuICA8cD5cbiAgICA8ZWwtaWNvbiBjb2xvcj1cInZhcigtLWVsLWNvbG9yLXByaW1hcnkpXCI+PEVsZW1lbnRQbHVzIC8+PC9lbC1pY29uPlxuICAgIEVsZW1lbnQgUGx1cyB7eyBFcFZlcnNpb24gfX0gKyBWdWUge3sgdnVlVmVyc2lvbiB9fVxuICA8L3A+XG48L3RlbXBsYXRlPlxuXG48c3R5bGU+XG4uY29udGFpbmVyIC5lbC1pbnB1dCB7XG4gIC0tZWwtaW5wdXQtaGVpZ2h0OiA0MnB4O1xufVxuPC9zdHlsZT5cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMDg0Ny1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTEwODQ3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTEwODQ3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTEwODQ3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

**Comment by u373693041** [comment]
Could this change be reviewed as soon as possible please? This bug has a great impact on me, thank you

**Comment by MrWeilian** [comment]
> Could this change be reviewed as soon as possible please? This bug has a great impact on me, thank you

@emojiiii @chenxch 大佬们

**Comment by chenxch** [comment]
[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJycpXG5jb25zdCB2YWwgPSByZWYoZmFsc2UpXG5jb25zdCBzZWxlY3REaXNhYmxlZCA9IHJlZihmYWxzZSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxkaXYgY2xhc3M9XCJjb250YWluZXJcIj5cbiAgICDlvIDlkK8v5YWz6Zetc2VsZWN0XG4gICAgPGVsLXN3aXRjaCB2LW1vZGVsPVwic2VsZWN0RGlzYWJsZWRcIiAvPlxuICAgIDxlbC1idXR0b24gQGNsaWNrPVwidmFsPSF2YWxcIj4gc2hvdyA8L2VsLWJ1dHRvbj5cbiAgICA8ZGl2IHYtc2hvdz1cInZhbFwiPlxuICAgICAgPGVsLXNlbGVjdCB2LW1vZGVsPVwibXNnXCIgOmRpc2FibGVkPVwic2VsZWN0RGlzYWJsZWRcIiBtdWx0aXBsZT5cbiAgICAgIDxlbC1vcHRpb24gdmFsdWU9XCIxXCIgbGFiZWw9XCIxMTExMTExMTExMVwiPjwvZWwtb3B0aW9uPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjIzMzMzMzMzMzNcIj48L2VsLW9wdGlvbj5cbiAgICA8L2VsLXNlbGVjdD5cbiAgICA8L2Rpdj5cbiAgICBcbiAgICAgICAgPGVsLXNlbGVjdCB2LW1vZGVsPVwibXNnXCIgOmRpc2FibGVkPVwic2VsZWN0RGlzYWJsZWRcIiBtdWx0aXBsZT5cbiAgICAgIDxlbC1vcHRpb24gdmFsdWU9XCIxXCIgbGFiZWw9XCIxMTExMTExMTExMVwiPjwvZWwtb3B0aW9uPlxuICAgICAgPGVsLW9wdGlvbiB2YWx1ZT1cIjJcIiBsYWJlbD1cIjIzMzMzMzMzMzNcIj48L2VsLW9wdGlvbj5cbiAgICA8L2VsLXNlbGVjdD5cbiAgICBcbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c3R5bGU+XG4gIC5jb250YWluZXIgLmVsLWlucHV0IHtcbiAgICAtLWVsLWlucHV0LWhlaWdodDogNDJweDtcbiAgfVxuPC9zdHlsZT5cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMDg0Ny1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTEwODQ3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTEwODQ3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTEwODQ3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)
There is also an initialization problem. I remember that there is a related pr, and I will look for it.

**Comment by chenxch** [comment]
#11374

File: packages/components/select/src/useSelect.ts [hunk 1]
```diff
@@ -101,6 +101,7 @@ export const useSelect = (props, states: States, ctx) => {
   const hoverOption = ref(-1)
   const queryChange = shallowRef<QueryChangeCtx>({ query: '' })
   const groupQueryChange = shallowRef('')
+  const originClientHeight = ref<number>(0)
```
Comment (chenxch) [inline packages/components/select/src/useSelect.ts:104]: `originClientHeight` does not use responsive declaration.

Comment (MrWeilian) [reply packages/components/select/src/useSelect.ts]: 你好，辛苦 review 啦。最新的提交把这个连同上面初始化问题的一起改了

**Review by chenxch — APPROVED** [review]
good job.

**Comment by MrWeilian** [comment]
> good job.

猴 我更新下分支

**Review by HeftyKoo — APPROVED** [review]

**Comment by MrWeilian** [comment]
😂 大佬们 这个pr能合并吗 又解决一次冲突啦🙈

**Pull request merged**